mkdocs_simple_plugin/plugin.py

```diff
@@ -467,27 +465,40 @@ def extract_from(self, from_directory, name, destination_directory):
         for item in self.semiliterate:
             name_match = item['pattern'].search(name)
             if name_match:
-                new_name = (name[:name_match.start(name_match.lastindex)] +
-                            '.md' +
-                            name[name_match.end(name_match.lastindex):])
                 if 'destination' in item:
                     new_name = name_match.expand(item['destination'])
+                else:
+                    if not name_match.lastindex:
+                        raise LookupError(
+                            "mkdocs-simple-plugin: No last group in match of" +
+                            "{} to {} and no destination".format(
+                                item['pattern'], name))
+                    new_name = (name[:name_match.start(name_match.lastindex)] +
+                                '.md' +
+                                name[name_match.end(name_match.lastindex):])
                 new_file = LazyFile(destination_directory, new_name)
                 with open(original) as original_file:
-                    extraction = StreamExtract(original_file,
-                                               new_file,
-                                               include_root=from_directory,
-                                               **item)
+                    utils.log.debug(
+                        "mkdocs-simple-plugin: Scanning {}...".format(original))
+                    productive = self.try_extraction(
+                        original_file, from_directory, new_file, **item)
                     new_file.close()
-                    if extraction.productive():
+                    if productive:
                         new_path = "{}/{}".format(destination_directory,
                                                   new_name)
                         utils.log.debug(
-                            "mkdocs-simple-plugin: {} --> {}".format(
-                                original, new_path))
+                            "        ... extracted {}".format(new_path))
                         new_paths.append((original, new_path))
         return new_paths
 
+    def try_extraction(self, original_file, root, new_file, **kwargs):
```

Comment:
I'm requesting to split this out as a separate function so that I can override just try_extraction in the semiliterate extension, since all I want to do is replace StreamExtract with a class that supports the inclusion syntax. That way I won't have to reiterate the entire definition of extract_from in the extension just to make that one change, which will make the extension significantly more resilient to changes in the simple plugin. Is it OK to keep try_extraction split out like this?

Reply:
Ah.  I'm fine with leaving it.

Although, fair warning that the Beyonce Rule applies here (if you liked it then you should've put a test on it).

Reply:
I absolutely understand your warning. Unfortunately I don't see how something like "the plugin has a method that allows the auxiliary class it is using to be overridden" could be tested in an integration test; it seems like it would require a unit test. But (a) introducing a unit testing framework to simple seems like a lot of work for the sake of just splitting out this method, and (b) would seem a little presumptuous for me to choose how this module would go about unit testing. I'll be happy to add a unit test for try_extraction if/when simple gets a framework for unit testing.

Reply:
It's not really a module that's meant to be inherited from, it's more of a warning that it may break in the future.

Reply:
I'm not trying to get you to sign a contract that you won't change the names of the SimplePlugin methods, etc. ;-) I am willing to adjust SemiliteratePlugin as needed to keep the inclusion syntax working.